src/Time/Slab.hpp

```diff
@@ -0,0 +1,125 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines class Slab
+
+#pragma once
+
+#include <functional>
+#include <limits>
+
+#include "ErrorHandling/Assert.hpp"
+
+class Time;
+class TimeDelta;
+
+namespace PUP {
+class er;
```

Comment:
It's the serialization class used by Charm++. I don't think we need elaboration on it in the code since it's inside the `PUP` namespace, though I'd be find with a comment along the lines of `// Charm++` somewhere here too